### PR TITLE
SO-3984: Add color highlighting to ECL editor

### PR DESCRIPTION
**Pull request by Henya14**
I am open to color suggestions. 
image

### CODE REVIEW

**Comment by Henya14** [comment]
image

File: com.b2international.snomed.ecl.ui/src/com/b2international/snomed/ecl/ui/EclUiModule.java [hunk 1]
```diff
@@ -53,4 +57,11 @@ public EclUiModule(AbstractUIPlugin plugin) {
 		return EclUiValidator.class;
 	}
 	
+	public Class<? extends AbstractAntlrTokenToAttributeIdMapper> bindAbstractAntlrTokenToAttributeIdMapper() {
+		return EclAntlrTokenToAttributeIdMapper.class ;
```
Comment (cmark) [inline com.b2international.snomed.ecl.ui/src/com/b2international/snomed/ecl/ui/EclUiModule.java:61]: Add these to the IDE module, please.

File: com.b2international.snomed.ecl.ui/src/com/b2international/snomed/ecl/ui/EclTokens.java [hunk 2]
```diff
@@ -17,45 +17,80 @@
...
 /**
+ * Tokens are made from the rules in the Ecl.xtext
  * @since 1.3.0
  */
 public class EclTokens {
 
 	public static final String PREFIX = "RULE_";
+	
+	public static final String SNOMED_IDENTIFIER = "SnomedIdentifier";
+	public static final String PIPE = "PIPE";
```
Comment (cmark) [inline com.b2international.snomed.ecl.ui/src/com/b2international/snomed/ecl/ui/EclTokens.java:29]: Would it be possible to reuse these from somewhere? I think Xtext generates these tokens already to a class.

**Review by cmark — CHANGES_REQUESTED** [review]

File: com.b2international.snomed.ecl.ui/src/com/b2international/snomed/ecl/ui/highlighting/EclSemanticHighlightingCalculator.java [hunk 3]
```diff
@@ -40,11 +40,15 @@ public void provideHighlightingFor(XtextResource resource, IHighlightedPositionA
 		for (INode abstractNode : allNodes) {
 			if (abstractNode.getGrammarElement() instanceof RuleCall) {
 				RuleCall rule = (RuleCall) abstractNode.getGrammarElement();
-
-				if (rule.getRule().getName().equals(EclTokens.CONCEPTID)) {
-					acceptor.addPosition(abstractNode.getOffset(), abstractNode.getLength(), EclTokens.CONCEPTID);
-				} else if (rule.getRule().getName().equals(EclTokens.TERM)) {
-					acceptor.addPosition(abstractNode.getOffset(), abstractNode.getLength(), EclTokens.TERM);
+				final String ruleName = rule.getRule().getName();
+				if (ruleName.equals(EclTokens.SNOMED_IDENTIFIER)) {
```
[the remark this answers is not in the document]
Comment (Henya14) [reply com.b2international.snomed.ecl.ui/src/com/b2international/snomed/ecl/ui/highlighting/EclSemanticHighlightingCalculator.java]: Done

File: com.b2international.snomed.ecl.ui/src/com/b2international/snomed/ecl/ui/EclUiModule.java (hunk 1 above)
Comment (Henya14) [reply com.b2international.snomed.ecl.ui/src/com/b2international/snomed/ecl/ui/EclUiModule.java:61]: Should I move every function from the UI module to the IDE module?

File: com.b2international.snomed.ecl.ui/src/com/b2international/snomed/ecl/ui/EclTokens.java (hunk 2 above)
Comment (Henya14) [reply com.b2international.snomed.ecl.ui/src/com/b2international/snomed/ecl/ui/EclTokens.java]: Done

File: com.b2international.snomed.ecl.ui/src/com/b2international/snomed/ecl/ui/EclUiModule.java (hunk 1 above)
Comment (cmark) [reply com.b2international.snomed.ecl.ui/src/com/b2international/snomed/ecl/ui/EclUiModule.java:69]: No, keep them there, @apeteri will move them if needed for the lang server development.

Comment (Henya14) [reply com.b2international.snomed.ecl.ui/src/com/b2international/snomed/ecl/ui/EclUiModule.java:61]: The ide module was removed. @cmark: ignore

**Review by cmark — APPROVED** [review]
LGTM! @apeteri could you please check if anything else is needed for this initial highlighting implementation (be it the highlighting itself or something for the LSP to work)? Thanks!

**Pull request merged**